packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -36,34 +36,86 @@ class ChartConnection extends DisposableController
   );
 
   Timer? _pollingTimer;
-  bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If completed, this instance was connected to the application.
+  final Completer<void> _initialized = Completer();
+
+  void _stopConnection() {
+    _pollingTimer?.cancel();
+    cancelStreamSubscriptions();
+    cancelListeners();
```

Comment:
these are called automatically by `dispose`, so I don't think we should be calling this here.

The reason I suggested to call these prior to adding the listeners / stream subscriptions is so that we do not add duplicate listeners and stream subs if the body of `maybeInitialize` will be called multiple times. You can see this pattern throughout the DevTools codebase when some initialization may occur in `didChangeDependencies`, for example, which can be called more than once.

If the body of `maybeInitialize` where the listeners / stream subs are added will not be executed more than once, then there is no need to call the cancel calls. However, given that they are a no-op when no listeners / stream subs are active, it does not hurt to call them before adding the listeners / stream subs, for robustness.

Reply:
I am cancelling them, because I want them to stop triggering functions, not because I want to dispose them.
They will not trigger events in case of disconnect. How about switching to offline mode?
Connection is still here and may trigger events.